### PR TITLE
ARROW-6477: [Packaging][Crossbow] Use Azure Pipelines to build linux packages

### PR DESCRIPTION
**Pull request by kszucs**
Also done some crossbow refactoring, need to update the wiki because the `sign` command has changed to `download-artficats`.

### CODE REVIEW

**Comment by kszucs** [comment]
@ursabot crossbow package centos-6

**Comment by kszucs** [comment]
@ursabot crossbow package centos-6

**Comment by kszucs** [comment]
@ursabot crossbow package centos-6

**Comment by kszucs** [comment]
@ursabot crossbow package centos-6

**Comment by kszucs** [comment]
@ursabot crossbow package centos-6

**Comment by kszucs** [comment]
@ursabot crossbow package centos-6

**Comment by kszucs** [comment]
@ursabot crossbow package centos-6

**Comment by kszucs** [comment]
@ursabot crossbow package centos-6

**Comment by kszucs** [comment]
@ursabot crossbow package centos-6 conda-linux-gcc-py37 conda-osx-clang-py37 conda-win-vs2015-py37

**Comment by kszucs** [comment]
@ursabot crossbow package -g conda -g linux

**Comment by kszucs** [comment]
@ursabot crossbow package -g conda -g linux

File: dev/tasks/tasks.yml [hunk 1]
```diff
@@ -433,7 +433,7 @@ tasks:
 
   # Don't use this for now. We can't complete this task in 50min on Travis CI.
   debian-stretch-arm64:
-    ci: travis
+    ci: azure
     platform: linux
     template: linux-packages/travis.linux.arm64.yml
```
Comment (kou) [inline dev/tasks/tasks.yml]: Can we enable this too?
We can't run this on Travis CI because it takes 50min+.

**Review by kou — COMMENTED** [review]
It seems that packager information isn't set correctly:

https://dev.azure.com/ursa-labs/crossbow/_build/results?buildId=818&view=logs&jobId=0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb&taskId=5b4cc83a-7bb0-5664-5bb1-588f7e4dc05b&lineStart=9389&lineEnd=9390&colStart=1&colEnd=1

```text
E: apache-arrow changes: changed-by-address-malformed <>
E: apache-arrow changes: changed-by-name-missing <>
E: apache-arrow changes: changed-by-address-missing <>
```

I'll push a commit to fix this.

**Comment by kou** [comment]
@ursabot crossbow package debian-buster

File: dev/tasks/tasks.yml (hunk 1 above)
Comment (kou) [reply dev/tasks/tasks.yml]: I'm trying this.

**Comment by kou** [comment]
@ursabot crossbow package debian-stretch-arm64

**Comment by kou** [comment]
@ursabot crossbow package debian-stretch-arm64

**Comment by kou** [comment]
Packager information has been solved.

**Comment by kou** [comment]
Wow. We can build debian-stretch-arm64 on Azure Pipelines. (It takes 2h 38m.)

@kszucs Should we fix conda-osx-clang-py27 failure in this pull request?

**Comment by kszucs** [comment]
@kou thanks for updating that! Almost three hours sounds pretty bad, but we can improve that in the future.

We can deal with the conda-osx-clang-py27 failure in another PR.

**Comment by kou** [comment]
OK! I'll merge this once CI is green.

**Review by kszucs — COMMENTED** [review]
+1, Thanks @kou!

**Pull request closed without merging**